File: system-test/java/src/test/java/com/optivem/eshop/systemtest/latest/contract/erp/BaseErpContractTest.java
package com.optivem.eshop.systemtest.latest.contract.erp;

import com.optivem.eshop.systemtest.latest.contract.base.BaseExternalSystemContractTest;
import org.junit.jupiter.api.Test;

public abstract class BaseErpContractTest extends BaseExternalSystemContractTest {
    @Test
    void shouldBeAbleToGetProduct() {
        scenario
                .given().product().withSku("SKU-123").withUnitPrice(12.0)
                .then().product("SKU-123").hasSku("SKU-123").hasPrice(12.0);
    }
}

File: system-test/typescript/tests/latest/contract/erp/BaseErpContractTest.ts
import type { ContractTest } from '../base/BaseExternalSystemContractTest.js';

export function registerErpContractTests(test: ContractTest): void {
  test('shouldBeAbleToGetProduct', async ({ scenario }) => {
    await scenario
      .given()
      .product()
      .withSku('SKU-123')
      .withUnitPrice(12.0)
      .then()
      .product('SKU-123')
      .hasSku('SKU-123')
      .hasPrice(12.0);
  });
}
